api-operator/pkg/mgw/virtualservice.go

```diff
@@ -121,13 +121,17 @@ func ValidateIstioConfigs(client *client.Client) error {
 	istioConfigs.GatewayName = istioConfigMap.Data[istioGatewayConfKey]
 
 	// host
-	if istioConfigMap.Data[istioHostConfKey] == "" {
+	// set host from API spec if given or from configmap
+	if api.Spec.IngressHostname != "" {
+		istioConfigs.Host = api.Spec.IngressHostname
+	} else if istioConfigMap.Data[istioHostConfKey] == "" {
 		err := errors.New("istio gateway host config is empty")
 		logVsc.Error(err, "Istio gateway host config is empty", "configmap", istioConfMapName,
 			"key", istioHostConfKey)
 		return err
+
+		istioConfigs.Host = istioConfigMap.Data[istioHostConfKey]
```

Comment:
This is fixed in https://github.com/wso2/k8s-api-operator/pull/413/files#diff-76b41b2d382ea90024cc26bfd84c7a4a

by adding else statement.